### DISCUSSION TITLE
[adonis 5] how could I change de meta data returned on paginate funcion?

### DISCUSSION THREAD

**Discussion by leognmotta** [adonisjs/core#1456] [Help]
I would like to make changes to this object:

```javascript
"meta": {
    "total": 1,
    "per_page": 15,
    "current_page": 1,
    "last_page": 1,
    "first_page": 1,
    "first_page_url": "/?page=1",
    "last_page_url": "/?page=1",
    "next_page_url": null,
    "previous_page_url": null
  }
```

**Comment by thetutlage** [comment]
You can construct the meta object yourself using the properties + methods available on the Pagination class instance.

**Reply by romileon** [reply]
Can you provide an example how to edit that object?

**Reply by Barbapapazes** [reply]
You can [use this](https://github.com/adonisjs/lucid/blob/efed38908680cca3b288d9b2a123586fab155b1d/src/Database/Paginator/SimplePaginator.ts#L20)!

**Reply by romileon** [reply]
thank you! I know is the SimplePaginator class, I was wondering how to construct the meta object without editing the class

**Reply by Barbapapazes** [reply]
You can extract data, create you custom object and replace it in the object before to return it to the client, all in the controller.

[This can inpire you](https://github.com/adonisjs/lucid/blob/6bfe2a5bd4937ff65d2d57456656eb5795dc3cc2/src/Orm/Paginator/index.ts#L19)
